sdk/cosmosdb/cosmos/src/queryExecutionContext/hybridQueryExecutionContext.ts

```diff
@@ -0,0 +1,432 @@
+  private componentsExecutionContext: ExecutionContext[] = [];
+  private pageSize: number;
+  private state: HybridQueryExecutionContextBaseStates;
+  private globalStatisticsAggregator: GlobalStatisticsAggregator;
+  private emitRawOrderByPayload: boolean = true;
+  private buffer: HybridSearchQueryResult[] = [];
+  private DEFAULT_PAGE_SIZE = 10;
+  private TOTAL_WORD_COUNT_PLACEHOLDER = "documentdb-formattablehybridsearchquery-totalwordcount";
+  private HIT_COUNTS_ARRAY_PLACEHOLDER = "documentdb-formattablehybridsearchquery-hitcountsarray";
+  private TOTAL_DOCUMENT_COUNT_PLACEHOLDER =
+    "documentdb-formattablehybridsearchquery-totaldocumentcount";
+  private RRF_CONSTANT = 60; // Constant for RRF score calculation
+
+  constructor(
+    private clientContext: ClientContext,
+    private collectionLink: string,
+    private options: FeedOptions,
+    private partitionedQueryExecutionInfo: PartitionedQueryExecutionInfo,
+    private correlatedActivityId: string,
+    private allPartitionsRanges: QueryRange[],
+  ) {
+    this.state = HybridQueryExecutionContextBaseStates.uninitialized;
+    this.pageSize = this.options.maxItemCount;
+    if (this.pageSize === undefined) {
+      this.pageSize = this.DEFAULT_PAGE_SIZE;
+    }
+    if (partitionedQueryExecutionInfo.hybridSearchQueryInfo.requiresGlobalStatistics) {
+      const globalStaticsQueryOptions: FeedOptions = { maxItemCount: this.pageSize };
+      this.globalStatisticsAggregator = new GlobalStatisticsAggregator();
+
+      const globalStatisticsQuery =
+        this.partitionedQueryExecutionInfo.hybridSearchQueryInfo.globalStatisticsQuery;
+      const globalStatisticsQueryExecutionInfo: PartitionedQueryExecutionInfo = {
+        partitionedQueryExecutionInfoVersion: 1,
+        queryInfo: {
+          distinctType: "None",
+          hasSelectValue: false,
+          groupByAliasToAggregateType: {},
+          rewrittenQuery: globalStatisticsQuery,
+          hasNonStreamingOrderBy: false,
+        },
+        queryRanges: this.allPartitionsRanges,
+      };
+
+      this.globalStatisticsExecutionContext = new ParallelQueryExecutionContext(
+        this.clientContext,
+        this.collectionLink,
+        globalStatisticsQuery,
+        globalStaticsQueryOptions,
+        globalStatisticsQueryExecutionInfo,
+        this.correlatedActivityId,
+      );
+    } else {
+      this.createComponentExecutionContexts();
+      this.state = HybridQueryExecutionContextBaseStates.initialized;
+    }
+  }
+  public async nextItem(diagnosticNode: DiagnosticNodeInternal): Promise<Response<any>> {
+    const nextItemRespHeaders = getInitialHeader();
+    while (
+      (this.state === HybridQueryExecutionContextBaseStates.uninitialized ||
+        this.state === HybridQueryExecutionContextBaseStates.initialized) &&
+      this.buffer.length === 0
+    ) {
+      await this.fetchMoreInternal(diagnosticNode, nextItemRespHeaders);
+    }
+
+    if (this.buffer.length > 0) {
```

Comment:
add condition for draining and done here along with buffer length

Reply:
addressed